packages/eas-cli/src/commandUtils/EasCommand.ts

```diff
@@ -12,26 +14,75 @@ import {
   initAsync as initAnalyticsAsync,
   logEvent,
 } from '../analytics/rudderstackClient';
-import { learnMore } from '../log';
+import Log, { learnMore } from '../log';
+import { interactiveOra, nonInteractiveOra } from '../ora';
 import { findProjectRootAsync } from '../project/projectUtils';
+import { InteractivePrompts, NonInteractivePrompts, Prompts } from '../prompts';
 import { getUserAsync } from '../user/User';
 import { ensureLoggedInAsync } from '../user/actions';
 import { easCliVersion } from '../utils/easCli';
 import { setVcsClient } from '../vcs';
 import GitClient from '../vcs/clients/git';
 
+type CommandOptions =
+  | {
+      nonInteractive: true;
+      json: true;
+    }
+  | {
+      nonInteractive: boolean;
+      json: false;
+    };
+
+export type CommandContext = Readonly<
+  CommandOptions & {
+    logger: Log;
+    prompts: Prompts;
+    ora: (options?: Options | string) => Ora;
+  }
+>;
+
 export default abstract class EasCommand extends Command {
+  static override globalFlags = {
+    json: Flags.boolean({
+      description: 'Enable JSON output, non-JSON messages will be printed to stderr.',
+      dependsOn: ['non-interactive'],
+    }),
+    'non-interactive': Flags.boolean({
+      description: 'Run the command in non-interactive mode.',
+    }),
+  };
+
+  // disable built in oclif json stuff
+  enableJsonFlag = false;
```

Comment:
```suggestion
  static enableJsonFlag = false;
```

docs suggest this is/should be static